File: app/Models/Friendship.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Friendship extends Model
{
    use HasFactory;

    public const STATUS_ACCEPTED = 'accepted';

    public const STATUS_PENDING = 'pending';

    protected $fillable = [
        'requester_id',
        'addressee_id',
        'status',
    ];

    public function requester(): BelongsTo
    {
        return $this->belongsTo(User::class, 'requester_id');
    }

    public function addressee(): BelongsTo
    {
        return $this->belongsTo(User::class, 'addressee_id');
    }

    public function involves(int $userId): bool
    {
        return $this->requester_id === $userId || $this->addressee_id === $userId;
    }
}

File: app/models/friendship.rb
class Friendship < ApplicationRecord
  STATUS_ACCEPTED = "accepted"
  STATUS_PENDING = "pending"

  belongs_to :requester, class_name: "User"
  belongs_to :addressee, class_name: "User"

  scope :pending, -> { where(status: STATUS_PENDING) }
  scope :accepted, -> { where(status: STATUS_ACCEPTED) }

  def involves?(user_id)
    requester_id == user_id || addressee_id == user_id
  end
end
